### PR TITLE
:broom: specify endpoint "v2" version in appropriate situations

### PR DESCRIPTION
**Pull request by ryandgoulding**

### CODE REVIEW

File: packages/ua-devtools/src/oapp/types.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ import type {
 import { ExecutorOptionType } from '@layerzerolabs/lz-v2-utilities'
 
 export interface IOApp extends IOmniSDK {
-    getEndpointSDK(): Promise<IEndpoint>
```
Comment (janjakubnanista) [inline packages/ua-devtools/src/oapp/types.ts:17]: I think we should keep this one as-is for now, until we decide what to do about interoperability

**Review by janjakubnanista — CHANGES_REQUESTED** [review]

File: packages/devtools-evm-hardhat/src/runtime.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { ConfigurationError } from './errors'
 import { HardhatContext } from 'hardhat/internal/context'
 import { Environment as HardhatRuntimeEnvironmentImplementation } from 'hardhat/internal/core/runtime-environment'
 import { EndpointId } from '@layerzerolabs/lz-definitions'
-import { EndpointBasedFactory, Factory, formatEid } from '@layerzerolabs/devtools'
+import { EndpointV2BasedFactory, Factory, formatEid } from '@layerzerolabs/devtools'
```
[the remark this answers is not in the document]
Comment (ryandgoulding) [reply packages/devtools-evm-hardhat/src/runtime.ts]: Done.

File: packages/devtools-evm-hardhat/src/runtime.ts [hunk 3]
```diff
@@ -120,7 +120,7 @@ export const getHreByNetworkName: GetByNetwork<HardhatRuntimeEnvironment> = pMem
  */
 export const createGetHreByEid = (
     hre = getDefaultRuntimeEnvironment()
-): EndpointBasedFactory<HardhatRuntimeEnvironment> =>
+): EndpointV2BasedFactory<HardhatRuntimeEnvironment> =>
```
[the remark this answers is not in the document]
Comment (ryandgoulding) [reply packages/devtools-evm-hardhat/src/runtime.ts]: Done.

File: packages/ua-devtools/src/oapp/types.ts (hunk 1 above)
Comment (ryandgoulding) [reply packages/ua-devtools/src/oapp/types.ts:17]: Done.

**Review by janjakubnanista — APPROVED** [review]

**Pull request merged**